src/Features/Core/Portable/PullMemberUp/AbstractPullMemberUpRefactoringProvider.cs

```diff
@@ -39,10 +41,9 @@ public override async Task ComputeRefactoringsAsync(CodeRefactoringContext conte
                 return;
             }
 
-            var allTargetClasses = FindAllTargetBaseClasses(userSelectNodeSymbol.ContainingType, context.Document.Project.Solution, context.CancellationToken);
-            var allTargetInterfaces = FindAllTargetInterfaces(userSelectNodeSymbol.ContainingType, context.Document.Project.Solution, context.CancellationToken);
+            var allTargets = FindAllValidTargets(userSelectNodeSymbol.ContainingType, context.Document.Project.Solution, context.CancellationToken);
 
-            if (allTargetInterfaces.Count() == 0 && allTargetClasses.Count() == 0)
+            if (allTargets.Count() == 0)
```

Comment:
.Count here makes it seem as ifthis is linq.  which means that this is lazily computing things.  which means we'll have to FindAllValidTargets multiple times.  Please return an ImmutableArray and use .Length here.